### ISSUE TITLE
🐞 routes that match a previously registered static file handler return a status code of 404

### ISSUE THREAD

**Issue by y21** [gofiber/fiber#250]
**Fiber version/commit**
Fiber v1.8.42

**Issue description**
I'm not sure if this is a bug or intended behavior, but routes that match a previously registered static file handler will 404 (unless `ctx.Status` is explicitly called). I am aware that this can easily be avoided by simply registering a handler before registering static file handlers, but I don't think it should set a status code of 404 when there *is* a handler that matches the route.
![image](https://user-images.githubusercontent.com/30553356/77847649-0aaca400-71bf-11ea-8b3f-0085d906c024.png)

**Expected behavior**
If the route matches one of the registered handlers, it should not set the status code to 404. If this is indeed intended behavior, the handler function should not be called.

**Steps to reproduce**
1) Run the code belolw
2) Visit localhost:3000
~> Webserver responds with `test`, but returns a status code of 404.

**Code snippet**

```go
package main

import (
  "github.com/gofiber/fiber"
)

func main() {
  app := fiber.New()

  app.Static("/", "./public/")
  app.Get("/", func (c *fiber.Ctx) {
    c.Send("test")
  })

  app.Listen(3000)
}
```

**Comment by Fenny** [comment]
@y21, will be fixed in `v1.8.43`!

**Comment by Fenny** [comment]
https://github.com/gofiber/fiber/releases/tag/v1.8.43
